toolchain/semantics/semantics_handle.cpp

```diff
@@ -372,6 +385,34 @@ auto SemanticsHandlePrefixOperator(SemanticsContext& context,
   return true;
 }
 
+auto SemanticsHandleQualifiedDeclaration(SemanticsContext& context,
+                                         ParseTree::Node parse_node) -> bool {
+  // The first two qualifiers in a chain will be a QualifiedDeclaration with two
+  // Identifier or expression children. Later qualifiers will have a
+  // QualifiedDeclaration as the first child, and an Identifier or expression as
+  // the second child.
+  //
+  // The child parse nodes are in reverse order versus how they need to be
+  // applied.
```

Comment:
I don't think I'm following this comment: my understanding is that given
```
fn Foo.Bar.Baz()
```
... the parse tree nodes are in this order
```
1: Foo
2:     Bar
3:    .
4:         Baz
5:        .
```
which looks like the order in which we apply them:

New --(`Foo`)-> Resolved --(`Bar`)-> Resolved --(`Baz`)-> Unresolved

Reply:
Rephrasing: "The nodes are applied in reverse order due to the postorder tree."